shell/src/main/java/alluxio/cli/bundler/command/CollectJvmInfoCommand.java

```diff
@@ -0,0 +1,153 @@
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.bundler.command;
+
+import alluxio.client.file.FileSystemContext;
+import alluxio.exception.AlluxioException;
+import alluxio.shell.CommandReturn;
+import alluxio.util.ShellUtils;
+import alluxio.util.SleepUtils;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Command that collects information about the JVMs.
+ * */
+public class CollectJvmInfoCommand extends AbstractInfoCollectorCommand {
+  public static final String COMMAND_NAME = "collectJvmInfo";
+  private static final Logger LOG = LoggerFactory.getLogger(CollectJvmInfoCommand.class);
+  private static final int COLLECT_JSTACK_TIMES = 3;
+  private static final int COLLECT_JSTACK_INTERVAL = 3 * 1000;
+
+  /**
+   * Creates an instance of {@link CollectJvmInfoCommand}.
+   *
+   * @param fsContext the {@link FileSystemContext} to execute in
+   * */
+  public CollectJvmInfoCommand(@Nullable FileSystemContext fsContext) {
+    super(fsContext);
+  }
+
+  @Override
+  public String getCommandName() {
+    return COMMAND_NAME;
+  }
+
+  @Override
+  public int run(CommandLine cl) throws AlluxioException, IOException {
+    // Determine the working dir path
+    mWorkingDirPath = getWorkingDirectory(cl);
+
+    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm-ss");
+    for (int i = 0; i < COLLECT_JSTACK_TIMES; i++) {
+      // Use time as output file name
+      LocalDateTime now = LocalDateTime.now();
+      String timeString = dtf.format(now);
+      LOG.info(String.format("Collecting JVM info at %s", timeString));
+
+      LOG.info("Checking current JPS");
+      Map<String, String> procs = getJps();
+      String jstackContent = dumpJstack(procs);
+
+      File outputFile = generateOutputFile(mWorkingDirPath,
+              String.format("%s-%s", getCommandName(), timeString));
+      FileUtils.writeStringToFile(outputFile, jstackContent);
+
+      // Interval
+      LOG.info(String.format("Wait for an interval of %s seconds", COLLECT_JSTACK_INTERVAL));
+      SleepUtils.sleepMs(LOG, COLLECT_JSTACK_INTERVAL);
+    }
+
+    return 0;
+  }
+
+  private Map<String, String> getJps() throws IOException {
+    Map<String, String> procs = new HashMap<>();
+
+    // Get Jps output
+    String[] jpsCommand = new String[]{"jps"};
```

Comment:
`jps` and `jstack` will only work if the Java process is started by the current user unless you are running command as super user. Is it possible to try running it using sudo first?

Reply:
Will this dump too many Java processes that are not necessarily relevant? For example there can be ZooKeeper, Yarn, HDFS and Spark (probably other apps).

My assumption is, this command is run by the one who started the Alluxio cluster. Do you think this is a valid assumption to make?

Reply:
It can be a different user than the alluxio service user. @ns1123 What do you think?

Reply:
Now the command attempts to do `sudo jps` and falls back to just `jps`.
When normal `jstack` fails on the pid, try `sudo jstack`. 
WRT to the cgroup constraint, I don't think that can be easily resolved in this. We can discuss further on this.